src/Interpreters/ProcessList.cpp

```diff
@@ -83,6 +83,7 @@ ProcessList::EntryPtr ProcessList::insert(const String & query_, const IAST * as
 
     {
         std::unique_lock lock(mutex);
+        OvercommitTrackerBlockerInThread overcommit_blocker; // To avoid deadlock in case of OOM
```

Comment:
`safeLock()`?

Reply:
Here we need to have `std::unique_lock`, because a little bit later conditional variable is used. I think it's better to leave this line this way.